src/librustc_mir/interpret/intrinsics/type_name.rs

```diff
@@ -226,3 +226,11 @@ pub fn type_name<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, ty: Ty<'tcx>) -> &'tcx t
         ty: tcx.mk_static_str(),
     })
 }
+
+/// Directly returns an `Allocation` containing an absolute path representation of the given type.
+pub(super)
+fn alloc_type_name<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, ty: Ty<'tcx>) -> &'tcx Allocation {
```

Comment:
please call this function in `type_name` above.

Also the indentation is a little weird, put the `pub(super) fn alloc_type_name<'a, 'tcx>(` on one line and the arguments on their own lines.

Reply:
Will do.

Reply:
I've made the changes. I'm not sure why `type_name.rs` is no longer auto-mergeable, when it was yesterday when I added `allloc_type_name`.

I won't be somewhere with access to a real checkout of my branch (i.e. not the web client) until later today, so if it can wait a while, I'll figure out what needs to be done to manually resolve the conflicts then.

Reply:
Ok, I got everything merged and changed, and it looks like the build is passing fine. I'll resolve this for now, let me know if there's anything else.